packages/grafana-ui/src/components/Segment/SegmentInput.tsx

```diff
@@ -34,9 +36,18 @@ export function SegmentInput<T>({
   if (!expanded) {
     return (
       <Label
+        disabled={disabled}
         Component={
           Component || (
-            <a className={cx('gf-form-label', 'query-part', !value && placeholder && 'query-placeholder', className)}>
+            <a
+              className={cx(
+                'gf-form-label',
+                'query-part',
+                !value && placeholder && 'query-placeholder',
+                getStyles<T>({ disabled: disabled }).link,
+                className
```

Comment:
fees like it would be better to use conditional classnames here: https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/components/Button/ToolbarButton.tsx#L42 
Then you can move the getStyles call to the top (dont pass any props), just return a style with a "disabled"   style on it. then compose all the classses using cx like

```typescript
cx(
 'gf-form-label',
  'query-part', {
     ['query-placeholder']: !value && placeholder,
     [styles.disabled]: disabled
  },
className);
```

Reply:
Thanks, I wasn't quite happy with how the styles were applied. This looks cleaner, I'll get on it.

Reply:
Good point @torkelo